pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/InstanceResponseBlock.java

```diff
@@ -41,6 +41,9 @@
   public InstanceResponseBlock(IntermediateResultsBlock intermediateResultsBlock) {
     try {
       _instanceResponseDataTable = intermediateResultsBlock.getDataTable();
+      long totalThreadCpuTimeNs = intermediateResultsBlock.getThreadCpuTimeNs();
+      _instanceResponseDataTable.getMetadata().put(DataTable.THREAD_CPU_TIME_NS_METADATA_KEY,
```

Comment:
Do we want to hold off on this until we also add the serialization part?

Reply:
I agree. Let's send the value from server to broker in the DataTable metadata (or whatever) only after we do the follow-up of measuring the data table serialization cost. cc @mqliang

Reply:
If we don't set the value as DataTable metadata here, then it's impossible to get the value in QueryScheduler.java, in other words, we should delete the ServerGauge emitting code also. I think at this moment, we can just put a log line here, and deleting the change in QueryScheduler.java. 

Another workaround is: add the value as DataTable metadata here, but in QueryScheduler.java, after writing log and emitting ServerGauge, `dataTable.getMetadata().remove(DataTable.THREAD_CPU_TIME_NS_METADATA_KEY)` so that this value will NOT be send back to broker (and add a TODO stating remove it when we have data table serialization cost measuring).

Any preference between the two option?

Reply:
I think logging should still be done in QueryScheduler since that's where all the metrics of a query are logged together in one line once the processing is done. So, we shouldn't log it in a new place now imo. 

As you suggested, I think we can do a put() here, retrieve the metric in QueryScheduler from dataTable metadata and then make sure to do a remove() before  `byte[] responseData = serializeDataTable(queryRequest, dataTable);` is called. Use the retrieved value for adding to table gauge and writing the log. For the remove(), add a TODO to revisit this when follow-up work of data table serialization cost measurement is done. Should be fine since we are going to do the follow-up immediately after this.

Reply:
done